frontend/public/components/nav.jsx

```diff
@@ -285,21 +285,25 @@ const imagestreamsStartsWith = ['imagestreams', 'imagestreamtags'];
 const monitoringAlertsStartsWith = ['monitoring/alerts', 'monitoring/alertrules'];
 const clusterSettingsStartsWith = ['settings/cluster', 'config.openshift.io'];
 
-const MonitoringNavSection_ = ({ urls }) => {
+const MonitoringNavSection_ = ({ flags, urls }) => {
   const prometheusURL = urls[MonitoringRoutes.Prometheus];
   const grafanaURL = urls[MonitoringRoutes.Grafana];
   const kibanaURL = urls[MonitoringRoutes.Kibana];
-  return window.SERVER_FLAGS.prometheusBaseURL || window.SERVER_FLAGS.alertManagerBaseURL || prometheusURL || grafanaURL || kibanaURL
+  const showAlerts = flags[FLAGS.CAN_GET_NS] && window.SERVER_FLAGS.prometheusBaseURL;
+  const showSilences = flags[FLAGS.CAN_GET_NS] && window.SERVER_FLAGS.alertManagerBaseURL;
+  return showAlerts || showSilences || prometheusURL || grafanaURL || kibanaURL
     ? <NavSection title="Monitoring">
-      {window.SERVER_FLAGS.prometheusBaseURL && <HrefLink href="/monitoring/alerts" name="Alerts" startsWith={monitoringAlertsStartsWith} />}
-      {window.SERVER_FLAGS.alertManagerBaseURL && <HrefLink href="/monitoring/silences" name="Silences" />}
+      {showAlerts && <HrefLink href="/monitoring/alerts" name="Alerts" startsWith={monitoringAlertsStartsWith} />}
+      {showSilences && <HrefLink href="/monitoring/silences" name="Silences" />}
       {prometheusURL && <ExternalLink href={prometheusURL} name="Metrics" />}
       {grafanaURL && <ExternalLink href={grafanaURL} name="Dashboards" />}
       {kibanaURL && <ExternalLink href={kibanaURL} name="Logging" />}
     </NavSection>
     : null;
 };
-const MonitoringNavSection = connectToURLs(MonitoringRoutes.Prometheus, MonitoringRoutes.Grafana, MonitoringRoutes.Kibana)(MonitoringNavSection_);
+const MonitoringNavSection = connectToFlags(FLAGS.CAN_GET_NS)(
+  connectToURLs(MonitoringRoutes.Prometheus, MonitoringRoutes.Grafana, MonitoringRoutes.Kibana)(MonitoringNavSection_)
```

Comment:
I'm not sure what the performance implications of connecting this component more than once are. It seems like it would be better to connect only once even if we can't use the `connectToFlags` and `connectToURLs` helpers.

Reply:
I refactored it to use a single `monitoringNavSectionStateToProps` function.

As far as I know, it's actually OK to double connect a component and we do that in a few places. However, I think it's probably sensible to be extra careful about performance in this case, since it's for the sidebar.